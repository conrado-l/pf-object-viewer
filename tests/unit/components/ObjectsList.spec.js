import Vue from 'vue'
import { mount } from '@vue/test-utils'
import ObjectsList from '@/views/ObjectsList'
import Vuetify from 'vuetify'
import { Store } from 'vuex-mock-store'

describe('ObjectsList.vue', () => {
  const mockupObjects = [
    {
      available: false,
      creation_date: 1553276681,
      description: 'An amazing blue shirt',
      id: '125',
      name: 'Shirt',
      type: 'clothing'
    },
    {
      available: true,
      creation_date: 1553274681,
      description: 'An amazing red shirt',
      id: '126',
      name: 'Shirt',
      type: 'clothing'
    }
  ]

  const store = new Store({
    state: {
      objectsList: { // Required for vuex-mock-store
        objects: mockupObjects
      }
    },
    getters: {
      'objectsList/getObjects': mockupObjects,
      'objectsList/getFilters': {
        byTerm: {
          search: 'pf',
          selected: 'id',
          options: []
        },
        byAvailability: {
          selected: 'yes',
          options: []
        }
      },
      'objectsList/getPagination': '',
      'objectsList/getSorting': {
        selected: 'id',
        options: []
      }
    }
  })

  const factoryMount = ({ route, router }) => {
    return mount(ObjectsList, {
      stubs: ['router-link', 'router-view'],
      mocks: {
        $store: store,
        $wait: {
          is: () => {
            return false
          }
        },
        $route: {
          ...route
        },
        $router: {
          ...router
        }
      }
      // Fix/workaround for Vuetify's warning about data-app
      // https://forum.vuejs.org/t/vuetify-data-app-true-and-problems-rendering-v-dialog-in-unit-tests/27495/9
      // attachToDocument: true
    })
  }

  beforeEach(() => {
    Vue.use(Vuetify) // Vuetify issue with createLocalVue: https://github.com/vuetifyjs/vuetify/issues/4964
    store.dispatch = jest.fn().mockResolvedValue('data') // dispatch has to return a promise
  })

  afterEach(() => {
    store.reset()
    // wrapper.destroy() // Should be called because of attachToDocument
  })

  it('should render the component correctly', () => {
    const wrapper = factoryMount({})

    expect(wrapper.element).toMatchSnapshot()
  })

  it('should call the action for hydrating with settings from the URL correctly', () => {
    const route = {
      query: {
        page: 1,
        search: 'pf',
        filterType: 'name',
        sortBy: 'id',
        available: 'yes'
      }
    }

    factoryMount({ route })

    expect(store.dispatch).toHaveBeenNthCalledWith(1, 'objectsList/applySettings', route.query)
  })

  it('should call the action for fetching the objects', () => {
    factoryMount({})

    expect(store.dispatch).toHaveBeenNthCalledWith(2, 'objectsList/fetchObjects')
  })

  it('should render the input values from the getters correctly', () => { // TODO: fix v-select not getting value
    const wrapper = factoryMount({})

    expect(wrapper.find('[data-test="input-search"').element.value).toBe('pf')
    // expect(wrapper.find('[data-test="input-filter-type"').element.value).toBe('name')
    // expect(wrapper.find('[data-test="input-filter-availability"').element.value).toBe('yes')
    // expect(wrapper.find('[data-test="input-sorting"').element.value).toBe('id')
  })

  it('should call updateRoute with the correct params on input', () => {
    const updateRoute = jest.fn()
    const wrapper = factoryMount({})

    wrapper.setMethods({ updateRoute })

    // wrapper.trigger() doesn't allow {target: {value: 'value'}}
    // https://vue-test-utils.vuejs.org/api/wrapper/trigger.html
    wrapper.find('[data-test="input-search"').element.value = 'fp'
    wrapper.find('[data-test="input-search"').trigger('input')

    wrapper.find('[data-test="input-filter-type"').element.value = 'name'
    wrapper.find('[data-test="input-filter-type"').trigger('input')

    wrapper.find('[data-test="input-filter-availability"').element.value = 'no'
    wrapper.find('[data-test="input-filter-availability"').trigger('input')

    wrapper.find('[data-test="input-sorting"').element.value = 'name'
    wrapper.find('[data-test="input-sorting"').trigger('input')

    expect(updateRoute).toHaveBeenNthCalledWith(1, 'search', 'fp')
    expect(updateRoute).toHaveBeenNthCalledWith(2, 'filterType', 'name')
    expect(updateRoute).toHaveBeenNthCalledWith(3, 'available', 'no')
    expect(updateRoute).toHaveBeenNthCalledWith(4, 'sortBy', 'name')
  })

  it('should update the route correctly when input updates', () => {
    const router = {
      push: jest.fn()
    }

    const route = {
      query: {
        page: 1,
        search: 'pf',
        filterType: 'name'
      }
    }

    const wrapper = factoryMount({ route, router })

    wrapper.vm.updateRoute('search', 'updatedPf')

    expect(router.push).toHaveBeenCalledTimes(1)
    expect(router.push).toHaveBeenCalledWith({
      name: 'objects-list',
      query: {
        filterType: 'name',
        page: 1,
        search: 'updatedPf'
      }
    })
  })

  it('should delete the route param when the input is falsy/empty', () => {
    const router = {
      push: jest.fn()
    }

    const route = {
      query: {
        page: 1,
        search: 'pf',
        filterType: 'name'
      }
    }

    const wrapper = factoryMount({ route, router })

    wrapper.vm.updateRoute('search', '')

    expect(router.push).toHaveBeenCalledTimes(1)
    expect(router.push).toHaveBeenCalledWith({
      name: 'objects-list',
      query: {
        filterType: 'name',
        page: 1
      }
    })
  })

  it('should call updateRoute with the correct params when pagination updates', () => {
    const updateRoute = jest.fn()
    const wrapper = factoryMount({})

    wrapper.setMethods({ updateRoute })

    wrapper.find('[data-test="input-pagination"').vm.$emit('input', 3) // Should trigger by click/change maybe

    expect(updateRoute).toHaveBeenCalledTimes(1)
    expect(updateRoute).toHaveBeenCalledWith('page', 3)
  })

  it('should return the objects array mapped for the table from the getObjects getter', () => {
    const wrapper = factoryMount({})

    expect(wrapper.vm.objects).toEqual([
      {
        'available': 'No',
        'description': 'An amazing blue shirt',
        'id': '125',
        'name': 'Shirt',
        'type': 'clothing'
      },
      {
        'available': 'Yes',
        'description': 'An amazing red shirt',
        'id': '126',
        'name': 'Shirt',
        'type': 'clothing'
      }
    ])
  })

  it('should return an empty array if getObjects getter is null', () => {
    const wrapper = factoryMount({})
    store.getters['objectsList/getObjects'] = null

    expect(wrapper.vm.objects).toEqual([])
  })

  // TODO: test table rendering correctly
  // TODO: test polling interval fetching objects
})
