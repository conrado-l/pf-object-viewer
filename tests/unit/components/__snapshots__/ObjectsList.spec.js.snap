// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectDetail.vue should render the component correctly 1`] = `
<div
  class="layout justify-center"
  data-app=""
>
  <div
    class="flex xs12 sm5 mx-2"
  >
    <div
      class="layout justify-center"
    >
      <div
        class="flex xs12 md4 mx-2"
      >
        <div
          class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-input--is-label-active v-input--is-dirty theme--light"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <input
                  aria-label="Search filter"
                  data-test="input-search"
                  name="search"
                  type="text"
                />
              </div>
              <div
                class="v-input__append-inner"
              >
                <div
                  class="v-input__icon v-input__icon--clear"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon v-icon--link material-icons theme--light"
                  >
                    clear
                  </i>
                </div>
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="flex xs12 md4 mx-2"
      >
        <div
          class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select theme--light"
          role="combobox"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-select__slot"
              >
                <label
                  aria-hidden="true"
                  class="v-label theme--light"
                  style="left: 0px; position: absolute;"
                >
                  Filter type
                </label>
                <div
                  class="v-select__selections"
                >
                  <input
                    aria-label="Filter type"
                    aria-readonly="false"
                    autocomplete="on"
                    data-test="input-filter-type"
                    name="filterType"
                    readonly="readonly"
                    type="text"
                  />
                </div>
                <div
                  class="v-input__append-inner"
                >
                  <div
                    class="v-input__icon v-input__icon--"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon v-icon--link material-icons theme--light"
                    />
                  </div>
                </div>
                <div
                  class="v-input__append-inner"
                >
                  <div
                    class="v-input__icon v-input__icon--append"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light"
                    >
                      arrow_drop_down
                    </i>
                  </div>
                </div>
              </div>
              <div
                class="v-menu"
              >
                <div
                  class="v-menu__content theme--light "
                  style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                >
                  <div
                    class="v-select-list v-card theme--light"
                  >
                    <div
                      class="v-list theme--light"
                      role="list"
                    >
                      <div
                        role="listitem"
                      >
                        <div
                          class="v-list__tile theme--light"
                        >
                          <div
                            class="v-list__tile__content"
                          >
                            <div
                              class="v-list__tile__title"
                            >
                              No data available
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="layout justify-center"
    >
      <div
        class="flex xs12 md4 mx-2"
      >
        <div
          class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select theme--light"
          role="combobox"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-select__slot"
              >
                <label
                  aria-hidden="true"
                  class="v-label theme--light"
                  style="left: 0px; position: absolute;"
                >
                  Available
                </label>
                <div
                  class="v-select__selections"
                >
                  <input
                    aria-label="Available"
                    aria-readonly="false"
                    autocomplete="on"
                    data-test="input-filter-availability"
                    name="available"
                    readonly="readonly"
                    type="text"
                  />
                </div>
                <div
                  class="v-input__append-inner"
                >
                  <div
                    class="v-input__icon v-input__icon--"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon v-icon--link material-icons theme--light"
                    />
                  </div>
                </div>
                <div
                  class="v-input__append-inner"
                >
                  <div
                    class="v-input__icon v-input__icon--append"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light"
                    >
                      arrow_drop_down
                    </i>
                  </div>
                </div>
              </div>
              <div
                class="v-menu"
              >
                <div
                  class="v-menu__content theme--light "
                  style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                >
                  <div
                    class="v-select-list v-card theme--light"
                  >
                    <div
                      class="v-list theme--light"
                      role="list"
                    >
                      <div
                        role="listitem"
                      >
                        <div
                          class="v-list__tile theme--light"
                        >
                          <div
                            class="v-list__tile__content"
                          >
                            <div
                              class="v-list__tile__title"
                            >
                              No data available
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="flex xs12 md6 mx-2"
      >
        <div
          class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select theme--light"
          role="combobox"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-select__slot"
              >
                <label
                  aria-hidden="true"
                  class="v-label theme--light"
                  style="left: 0px; position: absolute;"
                >
                  Sort by
                </label>
                <div
                  class="v-select__selections"
                >
                  <input
                    aria-label="Sort by"
                    aria-readonly="false"
                    autocomplete="on"
                    data-test="input-sorting"
                    name="sortBy"
                    readonly="readonly"
                    type="text"
                  />
                </div>
                <div
                  class="v-input__append-inner"
                >
                  <div
                    class="v-input__icon v-input__icon--"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon v-icon--link material-icons theme--light"
                    />
                  </div>
                </div>
                <div
                  class="v-input__append-inner"
                >
                  <div
                    class="v-input__icon v-input__icon--append"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light"
                    >
                      arrow_drop_down
                    </i>
                  </div>
                </div>
              </div>
              <div
                class="v-menu"
              >
                <div
                  class="v-menu__content theme--light "
                  style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                >
                  <div
                    class="v-select-list v-card theme--light"
                  >
                    <div
                      class="v-list theme--light"
                      role="list"
                    >
                      <div
                        role="listitem"
                      >
                        <div
                          class="v-list__tile theme--light"
                        >
                          <div
                            class="v-list__tile__content"
                          >
                            <div
                              class="v-list__tile__title"
                            >
                              No data available
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="v-card v-sheet theme--light"
    >
      <div
        class="v-card__title headline"
      >
        
        Objects List
      
      </div>
       
      <div
        class="spacer"
      />
       
      <div>
        <div
          class="v-table__overflow"
        >
          <table
            class="v-datatable v-table theme--light"
          >
            <thead>
              <tr>
                <th
                  aria-label="ID: Not sorted."
                  aria-sort="none"
                  class="column text-xs-left"
                  role="columnheader"
                  scope="col"
                >
                  ID
                </th>
                <th
                  aria-label="Name: Not sorted."
                  aria-sort="none"
                  class="column text-xs-left"
                  role="columnheader"
                  scope="col"
                >
                  Name
                </th>
                <th
                  aria-label="Description: Not sorted."
                  aria-sort="none"
                  class="column text-xs-left"
                  role="columnheader"
                  scope="col"
                >
                  Description
                </th>
                <th
                  aria-label="Type: Not sorted."
                  aria-sort="none"
                  class="column text-xs-left"
                  role="columnheader"
                  scope="col"
                >
                  Type
                </th>
                <th
                  aria-label="Available: Not sorted."
                  aria-sort="none"
                  class="column text-xs-left"
                  role="columnheader"
                  scope="col"
                >
                  Available
                </th>
              </tr>
              <tr
                class="v-datatable__progress"
              >
                <th
                  class="column"
                  colspan="5"
                />
              </tr>
            </thead>
            <tbody>
              <router-link-stub
                data-test="item-125"
                tag="tr"
                title="Go to detail"
                to="[object Object]"
              >
                <td>
                  125
                </td>
                <td>
                  Shirt
                </td>
                <td>
                  An amazing blue shirt
                </td>
                <td>
                  clothing
                </td>
                <td>
                  No
                </td>
              </router-link-stub>
              <router-link-stub
                data-test="item-126"
                tag="tr"
                title="Go to detail"
                to="[object Object]"
              >
                <td>
                  126
                </td>
                <td>
                  Shirt
                </td>
                <td>
                  An amazing blue shirt
                </td>
                <td>
                  clothing
                </td>
                <td>
                  No
                </td>
              </router-link-stub>
            </tbody>
          </table>
        </div>
      </div>
    </div>
     
    <div
      class="layout align-center justify-center my-2"
    >
      <ul
        class="v-pagination v-pagination--circle theme--light"
      >
        <li>
          <button
            class="v-pagination__navigation v-pagination__navigation--disabled"
            type="button"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              chevron_left
            </i>
          </button>
        </li>
        <li>
          <button
            class="v-pagination__navigation v-pagination__navigation--disabled"
            type="button"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              chevron_right
            </i>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`ObjectsList.vue should render the component correctly 1`] = `
<div
  class="layout justify-center"
  data-app=""
>
  <div
    class="flex xs12 sm5 mx-2"
  >
    <div
      class="layout justify-center"
    >
      <div
        class="flex xs12 md4 mx-2"
      >
        <div
          class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-input--is-label-active v-input--is-dirty theme--light"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <input
                  aria-label="Search filter"
                  data-test="input-search"
                  name="search"
                  type="text"
                />
              </div>
              <div
                class="v-input__append-inner"
              >
                <div
                  class="v-input__icon v-input__icon--clear"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon v-icon--link material-icons theme--light"
                  >
                    clear
                  </i>
                </div>
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="flex xs12 md4 mx-2"
      >
        <div
          class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select theme--light"
          role="combobox"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-select__slot"
              >
                <label
                  aria-hidden="true"
                  class="v-label theme--light"
                  style="left: 0px; position: absolute;"
                >
                  Filter type
                </label>
                <div
                  class="v-select__selections"
                >
                  <input
                    aria-label="Filter type"
                    aria-readonly="false"
                    autocomplete="on"
                    data-test="input-filter-type"
                    name="filterType"
                    readonly="readonly"
                    type="text"
                  />
                </div>
                <div
                  class="v-input__append-inner"
                >
                  <div
                    class="v-input__icon v-input__icon--"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon v-icon--link material-icons theme--light"
                    />
                  </div>
                </div>
                <div
                  class="v-input__append-inner"
                >
                  <div
                    class="v-input__icon v-input__icon--append"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light"
                    >
                      arrow_drop_down
                    </i>
                  </div>
                </div>
              </div>
              <div
                class="v-menu"
              >
                <div
                  class="v-menu__content theme--light "
                  style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                >
                  <div
                    class="v-select-list v-card theme--light"
                  >
                    <div
                      class="v-list theme--light"
                      role="list"
                    >
                      <div
                        role="listitem"
                      >
                        <div
                          class="v-list__tile theme--light"
                        >
                          <div
                            class="v-list__tile__content"
                          >
                            <div
                              class="v-list__tile__title"
                            >
                              No data available
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="layout justify-center"
    >
      <div
        class="flex xs12 md4 mx-2"
      >
        <div
          class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select theme--light"
          role="combobox"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-select__slot"
              >
                <label
                  aria-hidden="true"
                  class="v-label theme--light"
                  style="left: 0px; position: absolute;"
                >
                  Available
                </label>
                <div
                  class="v-select__selections"
                >
                  <input
                    aria-label="Available"
                    aria-readonly="false"
                    autocomplete="on"
                    data-test="input-filter-availability"
                    name="available"
                    readonly="readonly"
                    type="text"
                  />
                </div>
                <div
                  class="v-input__append-inner"
                >
                  <div
                    class="v-input__icon v-input__icon--"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon v-icon--link material-icons theme--light"
                    />
                  </div>
                </div>
                <div
                  class="v-input__append-inner"
                >
                  <div
                    class="v-input__icon v-input__icon--append"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light"
                    >
                      arrow_drop_down
                    </i>
                  </div>
                </div>
              </div>
              <div
                class="v-menu"
              >
                <div
                  class="v-menu__content theme--light "
                  style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                >
                  <div
                    class="v-select-list v-card theme--light"
                  >
                    <div
                      class="v-list theme--light"
                      role="list"
                    >
                      <div
                        role="listitem"
                      >
                        <div
                          class="v-list__tile theme--light"
                        >
                          <div
                            class="v-list__tile__content"
                          >
                            <div
                              class="v-list__tile__title"
                            >
                              No data available
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="flex xs12 md6 mx-2"
      >
        <div
          class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select theme--light"
          role="combobox"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-select__slot"
              >
                <label
                  aria-hidden="true"
                  class="v-label theme--light"
                  style="left: 0px; position: absolute;"
                >
                  Sort by
                </label>
                <div
                  class="v-select__selections"
                >
                  <input
                    aria-label="Sort by"
                    aria-readonly="false"
                    autocomplete="on"
                    data-test="input-sorting"
                    name="sortBy"
                    readonly="readonly"
                    type="text"
                  />
                </div>
                <div
                  class="v-input__append-inner"
                >
                  <div
                    class="v-input__icon v-input__icon--"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon v-icon--link material-icons theme--light"
                    />
                  </div>
                </div>
                <div
                  class="v-input__append-inner"
                >
                  <div
                    class="v-input__icon v-input__icon--append"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light"
                    >
                      arrow_drop_down
                    </i>
                  </div>
                </div>
              </div>
              <div
                class="v-menu"
              >
                <div
                  class="v-menu__content theme--light "
                  style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                >
                  <div
                    class="v-select-list v-card theme--light"
                  >
                    <div
                      class="v-list theme--light"
                      role="list"
                    >
                      <div
                        role="listitem"
                      >
                        <div
                          class="v-list__tile theme--light"
                        >
                          <div
                            class="v-list__tile__content"
                          >
                            <div
                              class="v-list__tile__title"
                            >
                              No data available
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="v-card v-sheet theme--light"
    >
      <div
        class="v-card__title headline"
      >
        
        Objects List
      
      </div>
       
      <div
        class="spacer"
      />
       
      <div>
        <div
          class="v-table__overflow"
        >
          <table
            class="v-datatable v-table theme--light"
          >
            <thead>
              <tr>
                <th
                  aria-label="ID: Not sorted."
                  aria-sort="none"
                  class="column text-xs-left"
                  role="columnheader"
                  scope="col"
                >
                  ID
                </th>
                <th
                  aria-label="Name: Not sorted."
                  aria-sort="none"
                  class="column text-xs-left"
                  role="columnheader"
                  scope="col"
                >
                  Name
                </th>
                <th
                  aria-label="Description: Not sorted."
                  aria-sort="none"
                  class="column text-xs-left"
                  role="columnheader"
                  scope="col"
                >
                  Description
                </th>
                <th
                  aria-label="Type: Not sorted."
                  aria-sort="none"
                  class="column text-xs-left"
                  role="columnheader"
                  scope="col"
                >
                  Type
                </th>
                <th
                  aria-label="Available: Not sorted."
                  aria-sort="none"
                  class="column text-xs-left"
                  role="columnheader"
                  scope="col"
                >
                  Available
                </th>
              </tr>
              <tr
                class="v-datatable__progress"
              >
                <th
                  class="column"
                  colspan="5"
                />
              </tr>
            </thead>
            <tbody>
              <router-link-stub
                data-test="item-125"
                tag="tr"
                title="Go to detail"
                to="[object Object]"
              >
                <td>
                  125
                </td>
                <td>
                  Shirt
                </td>
                <td>
                  An amazing blue shirt
                </td>
                <td>
                  clothing
                </td>
                <td>
                  No
                </td>
              </router-link-stub>
              <router-link-stub
                data-test="item-126"
                tag="tr"
                title="Go to detail"
                to="[object Object]"
              >
                <td>
                  126
                </td>
                <td>
                  Shirt
                </td>
                <td>
                  An amazing red shirt
                </td>
                <td>
                  clothing
                </td>
                <td>
                  Yes
                </td>
              </router-link-stub>
            </tbody>
          </table>
        </div>
      </div>
    </div>
     
    <div
      class="layout align-center justify-center my-2"
    >
      <ul
        class="v-pagination v-pagination--circle theme--light"
        data-test="input-pagination"
      >
        <li>
          <button
            class="v-pagination__navigation v-pagination__navigation--disabled"
            type="button"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              chevron_left
            </i>
          </button>
        </li>
        <li>
          <button
            class="v-pagination__navigation v-pagination__navigation--disabled"
            type="button"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              chevron_right
            </i>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
